docs/source/en/basics/router.md

```diff
@@ -0,0 +1,328 @@
+
+## How to Define Router
+
+- Define the routing rule in `app/router.js` file
+
+```js
+// app/router.js
+module.exports = app => {
+  app.get('/user/:id', 'user.info');
+};
+```
+
+- Implement the Controller in `app/controller` directory
+
+```js
+// app/controller/user.js
+exports.info = function* (ctx) {
+  ctx.body = {
+    name: `hello ${ctx.params.id}`,
+  };
+};
+```
+This simplest Router is done by now, when users do the request `GET /user/123`, the info function in `user.js` will be invoked.
+
+## Router config in detail
+
+Below is the complete definition of router, parameters can be determined depending on different scenes.
+
+```js
+app.verb('path-match', 'controller.action');
+app.verb('router-name', 'path-match', 'controller.action');
+app.verb('path-match', middleware1, ..., middlewareN, 'controller.action');
+app.verb('router-name', 'path-match', middleware1, ..., middlewareN, 'controller.action');
+```
+The complete definition of router includes 5 major parts:
+
+- verb - actions that users trigger, including get, post and so on, and will be explained in detail later.
+  * app.head - HEAD
+  * app.options - OPTIONS
+  * app.get - GET
+  * app.put - PUT
+  * app.post - POST
+  * app.patch - PATCH
+  * app.delete - DELETE
+  * app.del - this is a alias method due to the reservation of delete.
+  * app.redirect - redirect the request URL. For example, the most common case is to redirect the request accessing the root directory to the homepage.
+- router-name define a alias for the route, and URL can be generated by helper method `pathFor` and `urlFor` provided by Helper. (Optional)
+- path-match - URL path of the route.
+- middleware1 - multiple Middlewares can be configured in Router. (Optional)
+- controller.action - it's a string, take care. The framework will find the Controller sharing the name under `app/controller`, then assign the process to the configured action method. The action can be omitted if the Controller exports a method directly.
+
+### IMPORTANT
+
+- multiple Middlewares can be configured to execute serially in Router definition
+- Controller must be defined under `app/controller` directory
+- multiple Controllers can be defined within one file, and the specific one can be specified in the form of `controller-filename.function-name` when writing the routing rule.
+
+Here are some examples of writing routing rules:
+
+```js
+app.get('/home', 'home');
+app.get('/user/:id', 'user.page');
+app.post('/admin', isAdmin, 'admin');
+app.post('user', '/user', isLoginUser, hasAdminPermission, 'user.create');
+```
+
+### RESTful style URL definition
+
+We provide `app.resources('router-name', 'path-match', 'controller-name')` to produce [CRUD](https://en.wikipedia.org/wiki/Create,_read,_update_and_delete) structures on a path for convenient if you prefer the RESTful style URL definition.
```

Comment:
'to produce ... for convient'
suggestion:
'to fastly generate ...' or 'to generate ... for convenience'